crates/bevy_core_pipeline/src/dof/mod.rs

```diff
@@ -0,0 +1,907 @@
+            mag_filter: FilterMode::Linear,
+            min_filter: FilterMode::Linear,
+            ..default()
+        });
+
+        DepthOfFieldGlobalBindGroupLayout {
+            color_texture_sampler: sampler,
+            layout,
+        }
+    }
+}
+
+/// Creates the bind group layouts for the depth of field effect that are
+/// specific to each view.
+pub fn prepare_depth_of_field_view_bind_group_layouts(
+    mut commands: Commands,
+    view_targets: Query<(Entity, &DepthOfFieldSettings)>,
+    msaa: Res<Msaa>,
+    render_device: Res<RenderDevice>,
+) {
+    for (view, dof_settings) in view_targets.iter() {
+        // Create the bind group layout for the passes that take one input.
+        let single_input = render_device.create_bind_group_layout(
+            Some("depth of field bind group layout (single input)"),
+            &BindGroupLayoutEntries::sequential(
+                ShaderStages::FRAGMENT,
+                (
+                    uniform_buffer::<ViewUniform>(true),
+                    if *msaa != Msaa::Off {
+                        texture_depth_2d_multisampled()
+                    } else {
+                        texture_depth_2d()
+                    },
+                    texture_2d(TextureSampleType::Float { filterable: true }),
+                ),
+            ),
+        );
+
+        // If needed, create the bind group layout for the second bokeh pass,
+        // which takes two inputs. We only need to do this if bokeh is in use.
+        let dual_input = match dof_settings.mode {
+            DepthOfFieldMode::Gaussian => None,
+            DepthOfFieldMode::Bokeh => Some(render_device.create_bind_group_layout(
+                Some("depth of field bind group layout (dual input)"),
+                &BindGroupLayoutEntries::sequential(
+                    ShaderStages::FRAGMENT,
+                    (
+                        uniform_buffer::<ViewUniform>(true),
+                        if *msaa != Msaa::Off {
+                            texture_depth_2d_multisampled()
+                        } else {
+                            texture_depth_2d()
+                        },
+                        texture_2d(TextureSampleType::Float { filterable: true }),
+                        texture_2d(TextureSampleType::Float { filterable: true }),
+                    ),
+                ),
+            )),
+        };
+
+        commands
+            .entity(view)
+            .insert(ViewDepthOfFieldBindGroupLayouts {
+                single_input,
+                dual_input,
+            });
+    }
+}
+
+/// Configures depth textures so that the depth of field shader can read from
+/// them.
+///
+/// By default, the depth buffers that Bevy creates aren't able to be bound as
+/// textures. The depth of field shader, however, needs to read from them. So we
+/// need to set the appropriate flag to tell Bevy to make samplable depth
+/// buffers.
+pub fn configure_depth_of_field_view_targets(
```

Comment:
Note for the future, nothing that needs to be done in this PR, but I've also needed to do something like that in a few places. We should probably consider just making this the default in the future instead of forcing every depth based effect to do this.